src/charm.py

```diff
@@ -148,6 +148,7 @@ def _on_alertmanager_change(self, _):
     def _on_ingress_changed(self, _):
         self._configure()
         self.loki_provider.update_endpoint(url=self._external_url)
+        self.grafana_source_provider.update_source(source_url=self._external_url)
```

Comment:
This is certainly an improvement but iiuc this will become sufficient only when https://github.com/canonical/traefik-k8s-operator/issues/78 is fixed?

Reply:
I mean, not really, for the most part. It doesn't _actually_ matter to Grafana whether the ingress is currently listening or whether it takes a while. The Grafana process (not even charm, actual workload) itself will handle marking it as available whenever it is.

The crucial part here is in making sure that the datasource configuration is updated, whether or not it's truly available yet or has to wait a bit.